packages/dx-grid-core/src/plugins/integrated-filtering/computeds.js

```diff
@@ -50,27 +58,34 @@ const filterHierarchicalRows = (rows, predicate, getRowLevelKey, isGroupRow) =>
 
 export const filteredRows = (
   rows,
-  filters,
+  filterExpr,
```

Comment:
filterExpression?

Reply:
The filteredRows function has the filtersExpression variable in scope. We need use another name.

Reply:
Try to refactor the method. It is too complex. Extract getPredicateFromFilter function, for example.

Reply:
Up